modules/accounts_state/src/accounts_state.rs

```diff
@@ -183,97 +186,85 @@ impl AccountsState {
-                                Some(addr) => match Address::from_string(addr) {
-                                    Ok(Address::Stake(StakeAddress {
-                                        payload: StakeAddressPayload::StakeKeyHash(hash),
-                                        ..
-                                    })) => match history.lock().await.current() {
-                                        Some(state) => match state.get_stake_state(&hash) {
-                                            Some(stake) => match serde_json::to_string(&stake) {
-                                                Ok(body) => RESTResponse::with_json(200, &body),
-                                                Err(error) => RESTResponse::with_text(
-                                                    500,
-                                                    &format!("{error:?}").to_string(),
-                                                ),
-                                            },
-                                            None => RESTResponse::with_text(
-                                                404,
-                                                "Stake address not found",
-                                            ),
-                                        },
-
-                                        None => RESTResponse::with_text(500, "No state"),
-                                    },
-                                    _ => RESTResponse::with_text(400, "Not a stake address"),
-                                },
-                                None => {
-                                    RESTResponse::with_text(400, "Stake address must be provided")
-                                }
-                            }
-                        }
-                        _ => {
-                            error!("Unexpected message type {:?}", message);
-                            RESTResponse::with_text(500, "Unexpected message in REST request")
-                        }
-                    };
-
-                    Arc::new(Message::RESTResponse(response))
+        handle_rest_with_parameter(context.clone(), &handle_single_stake_topic, move |param| {
+            let history = history_stake_single.clone();
+            let param = param.to_string();
+
+            async move {
+                match Address::from_string(&param) {
+                    Ok(Address::Stake(StakeAddress {
+                        payload: StakeAddressPayload::StakeKeyHash(hash),
+                        ..
+                    })) => match history.lock().await.current() {
+                        Some(state) => match state.get_stake_state(&hash) {
+                            Some(stake) => match serde_json::to_string(&stake) {
+                                Ok(body) => Ok(RESTResponse::with_json(200, &body)),
+                                Err(error) => Err(anyhow!("{:?}", error)),
+                            },
```

Comment:
Not that this is a huge deal, but REST handlers would probably be nicer to write if `RESTResponse::with_json` took a `T: Serialize` instead of making the caller handle that.
```suggestion
                            Some(stake) => RESTResponse::with_json(200, &stake)
```